<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Russell Bainer" />

<meta name="date" content="2016-08-01" />

<title>Example Workflow For Processing a CRISPR-based Pooled Screen</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Example Workflow For Processing a CRISPR-based Pooled Screen</h1>
<h4 class="author"><em>Russell Bainer</em></h4>
<h4 class="date"><em>2016-08-01</em></h4>



<div id="example-workflow" class="section level3">
<h3>Example Workflow</h3>
<p>This is an example workflow for processing a pooled CRISPR-based screen using the provided sample data. See the various manpages for additional visualization options and algorithmic details.</p>
<div id="load-dependencies-and-data" class="section level5">
<h5>Load dependencies and data</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Biobase)
<span class="kw">library</span>(limma)
<span class="kw">library</span>(gCrisprTools)

<span class="kw">data</span>(<span class="st">&quot;es&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)
<span class="kw">data</span>(<span class="st">&quot;ann&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)
<span class="kw">data</span>(<span class="st">&quot;aln&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</code></pre></div>
</div>
<div id="make-a-sample-key-structured-as-a-factor-with-control-samples-in-the-first-level" class="section level5">
<h5>Make a sample key, structured as a factor with control samples in the first level</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sk &lt;-<span class="st"> </span><span class="kw">relevel</span>(<span class="kw">as.factor</span>(<span class="kw">pData</span>(es)$TREATMENT_NAME), <span class="st">&quot;ControlReference&quot;</span>)
<span class="kw">names</span>(sk) &lt;-<span class="st"> </span><span class="kw">row.names</span>(<span class="kw">pData</span>(es))</code></pre></div>
</div>
<div id="generate-a-contrast-of-interest-using-voomlimma" class="section level5">
<h5>Generate a contrast of interest using voom/limma</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(~<span class="st"> </span><span class="dv">0</span> +<span class="st"> </span>TREATMENT_NAME, <span class="kw">pData</span>(es))
<span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">'TREATMENT_NAME'</span>, <span class="st">''</span>, <span class="kw">colnames</span>(design))
contrasts &lt;-<span class="kw">makeContrasts</span>(DeathExpansion -<span class="st"> </span>ControlExpansion, <span class="dt">levels =</span> design)</code></pre></div>
</div>
<div id="optionally-trim-of-trace-reads-from-the-unnormalized-object-see-man-page-for-details" class="section level5">
<h5>Optionally, trim of trace reads from the unnormalized object (see man page for details)</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">es &lt;-<span class="st"> </span><span class="kw">ct.filterReads</span>(es, <span class="dt">trim =</span> <span class="dv">1000</span>, <span class="dt">sampleKey =</span> sk)</code></pre></div>
</div>
<div id="normalize-convert-to-a-voom-object-and-generate-a-contrast" class="section level5">
<h5>Normalize, convert to a voom object, and generate a contrast</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">es &lt;-<span class="st"> </span><span class="kw">ct.normalizeGuides</span>(es, <span class="dt">method =</span> <span class="st">&quot;scale&quot;</span>, <span class="dt">plot.it =</span> <span class="ot">TRUE</span>) <span class="co">#See man page for other options</span>
vm &lt;-<span class="st"> </span><span class="kw">voom</span>(<span class="kw">exprs</span>(es), design)

fit &lt;-<span class="st"> </span>limma::<span class="kw">lmFit</span>(vm, design)
fit &lt;-<span class="st"> </span>limma::<span class="kw">contrasts.fit</span>(fit, contrasts)
fit &lt;-<span class="st"> </span>limma::<span class="kw">eBayes</span>(fit)</code></pre></div>
</div>
<div id="edit-the-annotation-file-if-you-used-ct.filterreads-above" class="section level5">
<h5>Edit the annotation file if you used <code>ct.filterReads</code> above</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ann &lt;-<span class="st"> </span><span class="kw">ct.prepareAnnotation</span>(ann, fit, <span class="dt">controls =</span> <span class="st">&quot;NoTarget&quot;</span>)</code></pre></div>
</div>
<div id="summarize-grna-signals-to-identify-target-genes-of-interest" class="section level5">
<h5>Summarize gRNA signals to identify target genes of interest</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resultsDF &lt;-
<span class="st">  </span><span class="kw">ct.generateResults</span>(
    fit,
    <span class="dt">annotation =</span> ann,
    <span class="dt">RRAalphaCutoff =</span> <span class="fl">0.1</span>,
    <span class="dt">permutations =</span> <span class="dv">1000</span>,
    <span class="dt">scoring =</span> <span class="st">&quot;combined&quot;</span>, 
    <span class="dt">permutation.seed =</span> <span class="dv">2</span>
  )</code></pre></div>
</div>
<div id="optionally-just-load-an-example-results-object-for-testing-purposes" class="section level5">
<h5>Optionally, just load an example results object for testing purposes</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;fit&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)
<span class="kw">data</span>(<span class="st">&quot;resultsDF&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</code></pre></div>
</div>
<div id="trim-out-reads-as-necessary" class="section level5">
<h5>Trim out reads as necessary</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit &lt;-<span class="st"> </span>fit[(<span class="kw">row.names</span>(fit) %in%<span class="st"> </span><span class="kw">row.names</span>(ann)),]
resultsDF &lt;-<span class="st"> </span>resultsDF[(<span class="kw">row.names</span>(resultsDF) %in%<span class="st"> </span><span class="kw">row.names</span>(ann)),]</code></pre></div>
</div>
<div id="crispr-specific-quality-control-and-visualization-tools-see-man-pages-for-details" class="section level5">
<h5>Crispr-specific quality control and visualization tools (see man pages for details):</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.alignmentChart</span>(aln, sk)
<span class="kw">ct.rawCountDensities</span>(es, sk)</code></pre></div>
</div>
<div id="visualize-grna-abundance-distributions" class="section level5">
<h5>Visualize gRNA abundance distributions</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.gRNARankByReplicate</span>(es, sk) 
<span class="kw">ct.gRNARankByReplicate</span>(es, sk, <span class="dt">annotation =</span> ann, <span class="dt">geneSymb =</span> <span class="st">&quot;NoTarget&quot;</span>)  <span class="co">#Show locations of NTC gRNAs</span></code></pre></div>
</div>
<div id="visualize-control-guide-behavior-across-conditions" class="section level5">
<h5>Visualize control guide behavior across conditions</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.viewControls</span>(es, ann, sk, <span class="dt">normalize =</span> <span class="ot">FALSE</span>)
<span class="kw">t.viewControls</span>(es, ann, sk, <span class="dt">normalize =</span> <span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="visualize-gc-bias-across-samples-or-within-an-experimental-contrast" class="section level5">
<h5>Visualize GC bias across samples, or within an experimental contrast</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.GCbias</span>(es, ann, sk)
<span class="kw">ct.GCbias</span>(fit, ann, sk)</code></pre></div>
</div>
<div id="view-most-variable-grnasgenes-as-of-sequencing-library" class="section level5">
<h5>View most variable gRNAs/Genes (as % of sequencing library)</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.stackGuides</span>(es,
               sk,
               <span class="dt">plotType =</span> <span class="st">&quot;gRNA&quot;</span>,
               <span class="dt">annotation =</span> ann,
               <span class="dt">nguides =</span> <span class="dv">40</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.stackGuides</span>(es, 
               sk, 
               <span class="dt">plotType =</span> <span class="st">&quot;Target&quot;</span>, 
               <span class="dt">annotation =</span> ann)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.stackGuides</span>(es,
               sk,
               <span class="dt">plotType =</span> <span class="st">&quot;Target&quot;</span>,
               <span class="dt">annotation =</span> ann,
               <span class="dt">subset =</span> <span class="kw">names</span>(sk)[<span class="kw">grep</span>(<span class="st">'Expansion'</span>, sk)])</code></pre></div>
</div>
<div id="view-a-cdf-of-genesguides" class="section level5">
<h5>View a CDF of genes/guides</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.guideCDF</span>(es, sk, <span class="dt">plotType =</span> <span class="st">&quot;gRNA&quot;</span>)
<span class="kw">ct.guideCDF</span>(es, sk, <span class="dt">plotType =</span> <span class="st">&quot;Target&quot;</span>, <span class="dt">annotation =</span> ann)</code></pre></div>
</div>
<div id="view-top-enricheddepleted-candidates" class="section level5">
<h5>View top enriched/depleted candidates</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.topTargets</span>(fit,
              resultsDF,
              ann,
              <span class="dt">ntargets =</span> <span class="dv">10</span>,
              <span class="dt">enrich =</span> <span class="ot">TRUE</span>)
<span class="kw">ct.topTargets</span>(fit,
              resultsDF,
              ann,
              <span class="dt">ntargets =</span> <span class="dv">10</span>,
              <span class="dt">enrich =</span> <span class="ot">FALSE</span>)</code></pre></div>
</div>
<div id="view-the-grna-behavior-of-grnas-targeting-a-particular-gene-of-interest" class="section level5">
<h5>View the gRNA behavior of gRNAs targeting a particular gene of interest</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ct.viewGuides</span>(<span class="st">&quot;Target1633&quot;</span>, fit, ann)
<span class="kw">ct.gRNARankByReplicate</span>(es, sk, <span class="dt">annotation =</span> ann, <span class="dt">geneSymb =</span> <span class="st">&quot;Target1633&quot;</span>)</code></pre></div>
</div>
<div id="view-ontological-enrichment-within-the-depletedenriched-targets" class="section level5">
<h5>View ontological enrichment within the depleted/enriched targets</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">enrichmentResults &lt;-
<span class="st">  </span><span class="kw">ct.PantherPathwayEnrichment</span>(
    resultsDF,
    <span class="dt">pvalue.cutoff =</span> <span class="fl">0.01</span>,
    <span class="dt">enrich =</span> <span class="ot">TRUE</span>,
    <span class="dt">organism =</span> <span class="st">'mouse'</span>
  )</code></pre></div>
</div>
<div id="test-a-gene-set-for-enrichment-within-target-candidates" class="section level5">
<h5>Test a gene set for enrichment within target candidates</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;essential.genes&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)
ROCs &lt;-<span class="st"> </span><span class="kw">ct.ROC</span>(resultsDF, essential.genes, <span class="dt">stat =</span> <span class="st">&quot;deplete.p&quot;</span>)
PRCs &lt;-<span class="st"> </span><span class="kw">ct.PRC</span>(resultsDF, essential.genes, <span class="dt">stat =</span> <span class="st">&quot;deplete.p&quot;</span>)</code></pre></div>
</div>
<div id="make-reports-in-a-directory-of-interest" class="section level5">
<h5>Make reports in a directory of interest</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">path2report &lt;-<span class="st">      </span><span class="co">#Make a report of the whole experiment</span>
<span class="st">  </span>gCrisprTools::<span class="kw">ct.makeReport</span>(<span class="dt">fit =</span> fit, 
                              <span class="dt">eset =</span> es, 
                              <span class="dt">sampleKey =</span> sk, 
                              <span class="dt">annotation =</span> ann, 
                              <span class="dt">results =</span> resultsDF, 
                              <span class="dt">aln =</span> aln, 
                              <span class="dt">outdir =</span> <span class="st">&quot;.&quot;</span>) 

path2QC &lt;-<span class="st">          </span><span class="co">#Or one focusing only on experiment QC</span>
<span class="st">  </span>gCrisprTools::<span class="kw">ct.makeQCReport</span>(es, 
                                <span class="dt">trim =</span> <span class="dv">1000</span>, 
                                <span class="dt">log2.ratio =</span> <span class="fl">0.05</span>, 
                                <span class="dt">sampleKey =</span> sk, 
                                <span class="dt">annotation =</span> ann, 
                                <span class="dt">aln =</span> aln, 
                                <span class="dt">identifier =</span> <span class="st">'Crispr_QC_Report'</span>,
                                <span class="dt">lib.size =</span> <span class="ot">NULL</span>
                                )                

path2Contrast &lt;-<span class="st">    </span><span class="co">#Or Contrast-specific one</span>
<span class="st">  </span>gCrisprTools::<span class="kw">ct.makeContrastReport</span>(<span class="dt">eset =</span> es, 
                                      <span class="dt">fit =</span> fit, 
                                      <span class="dt">sampleKey =</span> sk, 
                                      <span class="dt">results =</span> resultsDF, 
                                      <span class="dt">annotation =</span> ann, 
                                      <span class="dt">comparison.id =</span> <span class="ot">NULL</span>, 
                                      <span class="dt">identifier =</span> <span class="st">'Crispr_Contrast_Report'</span>)            </code></pre></div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
